src/app/(session)/session/[id]/_components/puzzles/memory-game.tsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+
+import { useState } from "react";
+import { useDroppable } from "@dnd-kit/core";
+import { PuzzleEnums } from "../../_types";
+
+export default function MemoryGame() {
+  const { setNodeRef } = useDroppable({
+    id: PuzzleEnums.SOUNDSTONES,
+  });
+  const solution = ["A4", "A4", "C2", "C2", "A4"];
+  const checkOrderWithSolution = (stone: string) => {
+    const checking = [...stoneOrder, stone];
+    if (solution[checking.length - 1] === checking[checking.length - 1]) {
+      console.log(checking);
+      if (solution.length === checking.length) {
+        setStoneOrder([]);
+        console.log("YOU WIN YIPPEE");
+      }
+      return;
+    } else {
+      console.log([]);
+      setStoneOrder([]);
+    }
+  };
+  type TouchStoneProps = {
+    stone: string;
+  };
+  const [stoneOrder, setStoneOrder] = useState<string[]>([]);
+  const touchStone = ({ stone }: TouchStoneProps) => {
+    setStoneOrder([...stoneOrder, stone]);
+    const audio = new Audio(`/audio/GrandPiano-${stone}-mf.wav`);
+    audio.play();
+    setTimeout(() => {
+      audio.pause();
+    }, 2000);
+    checkOrderWithSolution(stone);
+  };
+
+  return (
+    <div
+      className="relative mx-auto flex h-64 w-64 justify-center"
+      ref={setNodeRef}
+    >
+      <div className="absolute left-1/2 top-0 -translate-x-1/2 transform">
+        <button
+          onClick={() => touchStone({ stone: "A4" })}
+          className="mushroom h-12 w-12 rounded-full bg-green-500"
+        ></button>
+      </div>
+      <div className="absolute left-full top-1/4 -translate-x-1/2 transform">
+        <button
+          onClick={() => touchStone({ stone: "C2" })}
+          className="mushroom h-12 w-12 rounded-full bg-green-500"
+        ></button>
+      </div>
+      <div className="absolute left-full top-1/2 -translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+      <div className="absolute left-full top-3/4 -translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+      <div className="absolute bottom-0 left-1/2 -translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+      <div className="absolute right-full top-3/4 translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+      <div className="absolute right-full top-1/2 translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+      <div className="absolute right-full top-1/4 translate-x-1/2 transform">
+        <button className="mushroom h-12 w-12 rounded-full bg-green-500"></button>
+      </div>
+    </div>
+  );
+}
```

Comment:
```suggestion
"use client";

import { useState } from "react";
import { useDroppable } from "@dnd-kit/core";
import { PuzzleEnums } from "../../_types";
import { cn } from "./utils"; // Adjust the import path as needed

const SOLUTION = ["A4", "A4", "C2", "C2", "A4"];
const STONE_POSITIONS = [
  { className: "top-0 left-1/2 -translate-x-1/2", stone: "A4" },
  { className: "top-1/4 left-full -translate-x-1/2", stone: "C2" },
  { className: "top-1/2 left-full -translate-x-1/2" },
  { className: "top-3/4 left-full -translate-x-1/2" },
  { className: "bottom-0 left-1/2 -translate-x-1/2" },
  { className: "top-3/4 right-full translate-x-1/2" },
  { className: "top-1/2 right-full translate-x-1/2" },
  { className: "top-1/4 right-full translate-x-1/2" },
];

export default function MemoryGame() {
  const [stoneOrder, setStoneOrder] = useState<string[]>([]);
  const { setNodeRef } = useDroppable({ id: PuzzleEnums.SOUNDSTONES });

  const checkSolution = (stone: string) => {
    const newOrder = [...stoneOrder, stone];
    if (SOLUTION[newOrder.length - 1] === stone) {
      if (SOLUTION.length === newOrder.length) {
        console.log("YOU WIN YIPPEE");
        setStoneOrder([]);
      } else {
        setStoneOrder(newOrder);
      }
    } else {
      setStoneOrder([]);
    }
  };

  const touchStone = (stone: string) => {
    const audio = new Audio(`/audio/GrandPiano-${stone}-mf.wav`);
    audio.play();
    setTimeout(() => audio.pause(), 2000);
    checkSolution(stone);
  };

  return (
    <div className={cn("relative mx-auto flex h-64 w-64 justify-center")} ref={setNodeRef}>
      {STONE_POSITIONS.map((pos, index) => (
        <div 
          key={index} 
          className={cn(
            "absolute",
            pos.className,
            pos.stone && "cursor-pointer hover:opacity-80"
          )}
        >
          <button
            onClick={() => pos.stone && touchStone(pos.stone)}
            className={cn(
              "mushroom h-12 w-12 rounded-full bg-green-500",
              !pos.stone && "opacity-50 cursor-not-allowed"
            )}
            disabled={!pos.stone}
          />
        </div>
      ))}
    </div>
  );
}
```


Not certain if this is 100% the right path but just something quick that I felt made it cleaner